src/main/kotlin/com/github/thinkami/railroads/ui/table/RoutesTable.kt

```diff
@@ -28,6 +29,17 @@ class RoutesTable: JBTable(), DataProvider {
      */
     init {
         addMouseListener(RailroadsPopupHandler())
+
+        addMouseListener(object : MouseAdapter() {
+            override fun mouseClicked(e: MouseEvent) {
+                if (e.clickCount == 2) {
+                    val route = getSelectedRoute()
+                    if (route != null && route.canNavigate()) {
+                        route.navigate(true)
+                    }
+                }
+            }
+        })

```

Comment:
While `DoubleClickListener` is poorly documented and somewhat difficult to understand, you can find it listed in the `Useful Classes` section on the following page:
https://plugins.jetbrains.com/docs/intellij/ui-faq.html#useful-classes

The actual class implementation and usage details can be found here:

- https://github.com/JetBrains/intellij-community/blob/0910bc0223a43ee25dd5f42743537138a63309e4/platform/platform-api/src/com/intellij/ui/DoubleClickListener.java
- https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/find/actions/ShowUsagesAction.java#L1065

That said, as I mentioned in my previous response, we could also proceed by having me make the required modifications.